clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -7471,6 +7471,175 @@ test_record_scheduler()
     check_next(stream0, record_scheduler_t::STATUS_EOF);
 }
 
+static void
+test_record_scheduler_i7574()
+{
+    // Demonstrates how the scheduler responds to traces with the i#7574 issue.
+    std::cerr << "\n----------------\nTesting record_scheduler_t for i#7574 workaround\n";
+    static constexpr memref_tid_t TID_A = 42;
+    static constexpr memref_tid_t TID_B = TID_A + 1;
+    static constexpr memref_tid_t PID_A = 142;
+    static constexpr memref_tid_t PID_B = PID_A + 1;
+    static constexpr int NUM_OUTPUTS = 1;
+    static constexpr addr_t ENCODING_SIZE = 2;
+    static constexpr addr_t ENCODING_IGNORE = 0xfeed;
+    static constexpr uint64_t INITIAL_TIMESTAMP_A = 10;
+    static constexpr uint64_t INITIAL_TIMESTAMP_B = 20;
+    std::vector<trace_entry_t> refs_A = {
+        /* clang-format off */
+        test_util::make_thread(TID_A),
+        test_util::make_pid(PID_A),
+        test_util::make_version(TRACE_ENTRY_VERSION),
+        test_util::make_timestamp(INITIAL_TIMESTAMP_A),
+        test_util::make_encoding(ENCODING_SIZE, ENCODING_IGNORE),
+        test_util::make_instr(10),
+        // Second instr, but the chunk end breaks up the
+        // branch_target marker and the instr (i#7574).
```

Comment:
So the wrong thing will happen in just one thread with -skip_instrs to the 2nd chunk?

Reply:
Yes, https://github.com/DynamoRIO/dynamorio/issues/7574#issue-3263694346 notes one such case: view_t would not see the branch_target if the prior chunk is skipped over.

The test added in this PR is a more important bug though.

Reply:
Harder to fix an ROI skip when -skip_instrs skips over the chunk that had the branch_target.

Only way for those traces would be to re-decompose.